transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala

```diff
@@ -136,25 +136,26 @@ object HcaPipelineBuilder extends PipelineBuilder[Args] {
     * @param entityType the metadata entity type to prepend for the id field
     * @param fileName the raw filename of the metadata file
     * @param metadata the content of the metadata file in Msg format
+    * @param descriptor the content of the descriptor file in Msg format
     * @return a Msg object in the desired output format
     */
-  def transformFileMetadata(entityType: String, fileName: String, metadata: Msg): Msg = {
+  def transformFileMetadata(
+    entityType: String,
+    fileName: String,
+    metadata: Msg,
+    descriptor: Msg
+  ): Msg = {
     val (entityId, entityVersion) = getEntityIdAndVersion(fileName)
-    val contentHash = metadata
-      .tryRead[String]("file_core", "file_provenance", "crc32c")
-      .getOrElse(metadata.read[String]("file_core", "crc32c"))
-    val dataFileName = metadata.read[String]("file_core", "file_name")
-    val (fileId, fileVersion) = getFileIdAndVersion(dataFileName)
+    val dataFileName = descriptor.read[String]("file_name")
     // put values in the form we want
     Obj(
       mutable.LinkedHashMap[Msg, Msg](
         Str(s"${entityType}_id") -> Str(entityId),
         Str("version") -> Str(entityVersion),
         Str("content") -> Str(encode(metadata)),
-        Str("crc32c") -> Str(contentHash),
-        Str("source_file_id") -> Str(fileId),
-        Str("source_file_version") -> Str(fileVersion),
-        Str("virtual_path") -> Str(s"/$dataFileName")
+        Str("virtual_path") -> Str(s"/$dataFileName"),
+        Str("crc32c") -> Str(descriptor.read[String]("crc32c")),
```

Comment:
```suggestion
        Str("crc32c") -> descriptor.read[Msg]("crc32c"),
```
Would pass through whatever value is stored in the field, with less object allocation / parsing. It wouldn't assert that the value is a string, though.